app/controllers/v2/profiles_controller.rb

```diff
@@ -20,7 +22,11 @@ def profiles
     end
 
     def profile
-      @profile ||= authorize(expand_resource.find(permitted_params[:id]))
+      @profile ||= if ::UUID.validate(permitted_params[:id])
```

Comment:
I would put this logic into the concern as it is repetitive in more than one controller.